robottelo/hosts.py

```diff
@@ -1752,14 +1752,26 @@ def __init__(self, hostname=None, **kwargs):
         self._api = type('api', (), {'_configured': False})
         self._cli = type('cli', (), {'_configured': False})
 
+    def _swap_nailgun(self, new_version):
```

Comment:
Would you like this to be a private function or public?

I assume this function would be called with the `upgrade_process` fixture(or whatever we will create for upgrades). Agree that `upgrade_process` could be part of the target
-sat but not necessary we need to tie `_swap_nailgun` method with satellite object it could be independent fixture or utils function, since its being operated on pytest host machine and not on satellite.